### ISSUE TITLE
Undefined table: field_revision_field_resource_titles

### ISSUE THREAD

**Issue by laceysanderson** [tripal/tripal#85]
Process: Upgrade from 2.1 => 3.0-beta3: content migration for a generic Tripal 2.1

Steps:
1. Choose to Migrate content (step2: admin/tripal/storage/chado/migrate) for Tripal v2 Analysis'
2. Run Tripal job and receive errors (see full output below).
3. Check interface, my single Analysis Tripal Entity has been created correctly...
4. Check the database, the table field_revision_field_resource_titles doesn't exist.

````
Tripal Job Launcher
Running as user 'XXX'
-------------------
Calling: tripal_chado_migrate_records(Array, 13)
Migrating selected Tripal v2 content...

NOTE: Populating of chado_cvterm_mapping table is performed using a database transaction. 
If the load fails or is terminated prematurely then the entire set of 
insertions/updates is rolled back and will not be found in the database

Examining analysis...
Examining contact...
Examining control...
Examining feature...
Examining featuremap...
Examining genotype...
Examining library...
Examining organism...
Examining organism...
Examining phenotype...
Examining phylotree...
Examining project...
Examining protocol...
Examining protocolparam...
Examining pub...
Examining stock...
Examining stockcollection...
Examining studyfactor...
Examining synonym...
Adding mapped tripal term: organism
Adding mapped tripal term: analysis
Adding mapped tripal term: project

Done.

NOTE: publishing records is performed using a database transaction. 
If the load fails or is terminated prematurely then the entire set of 
is rolled back with no changes to the database

WD trpentity: The URL alias, analysis/1, already exists for another page. Please ensure the pattern supplied on the Analysis Edit Page under URL Path options is  [warning]
unique.
[site http://default] [TRIPAL WARNING] [TRPENTITY] The URL alias, analysis/1, already exists for another page. Please ensure the pattern supplied on the <a href="/admin/structure/bio_data/manage/bio_data_2" target="_blank">Analysis Edit Page</a> under URL Path options is unique.
WD php: PDOException: SQLSTATE[42P01]: Undefined table: 7 ERROR:  relation "field_revision_field_resource_titles" does not exist                                  [error]
LINE 18:       FROM field_revision_field_resource_titles RT
                    ^: 
      SELECT
        entity_id,
        max(revision_id) AS vid,
        delta,
        (SELECT field_resource_titles_value
         FROM field_revision_field_resource_titles
         WHERE entity_id = RT.entity_id
         AND revision_id = max(RT.revision_id)
         AND delta = RT.delta
        ),
        (SELECT field_resource_blocks_value
         FROM field_revision_field_resource_blocks
         WHERE entity_id = RT.entity_id
         AND revision_id = max(RT.revision_id)
         AND delta = RT.delta
        )
      FROM field_revision_field_resource_titles RT
      WHERE RT.entity_id = :nid
      GROUP BY entity_id, delta
      ORDER BY RT.delta
    ; Array
(
    [:nid] => 2
)
 in tripal_chado_migrate_resource_blocks() (line 971 of /var/www/dev/trp-upgrade/sites/all/modules/tripal/tripal_chado/includes/tripal_chado.migrate.inc).
Cannot modify header information - headers already sent by (output started at /usr/share/php/drush/includes/output.inc:37) bootstrap.inc:1481                     [warning]
PDOException: SQLSTATE[42P01]: Undefined table: 7 ERROR:  relation "field_revision_field_resource_titles" does not exist
LINE 18:       FROM field_revision_field_resource_titles RT
                    ^: 
      SELECT
        entity_id,
        max(revision_id) AS vid,
        delta,
        (SELECT field_resource_titles_value
         FROM field_revision_field_resource_titles
         WHERE entity_id = RT.entity_id
         AND revision_id = max(RT.revision_id)
         AND delta = RT.delta
        ),
        (SELECT field_resource_blocks_value
         FROM field_revision_field_resource_blocks
         WHERE entity_id = RT.entity_id
         AND revision_id = max(RT.revision_id)
         AND delta = RT.delta
        )
      FROM field_revision_field_resource_titles RT
      WHERE RT.entity_id = :nid
      GROUP BY entity_id, delta
      ORDER BY RT.delta
    ; Array
(
    [:nid] => 2
)
 in tripal_chado_migrate_resource_blocks() (line 971 of /var/www/dev/trp-upgrade/sites/all/modules/tripal/tripal_chado/includes/tripal_chado.migrate.inc).
Drush command terminated abnormally due to an unrecoverable error.
````

**Comment by laceysanderson** [comment]
Repeatable for all content types...

**Comment by spficklin** [comment]
Are you using the most recent development version or the beta3 version?

**Comment by laceysanderson** [comment]
I was as of the posting of this issue. I always check the most recent dev before posting an issue :-)

**Comment by chunhuaicheng** [comment]
Hi, I added a check to skip the migration of the resource blocks/titles/links if a site wasn't set up for these. Could you check if this fixes the bug please?

**Comment by laceysanderson** [comment]
Confirmed this issue is fixed in the most recent dev version (3df04d61728ae8527db892def40d3479cb6a8e78).
